modules/integration/tests-ui-integration/src/test/java/org/wso2/carbon/mb/ui/test/dlc/DLCDurableTopicTestCase.java

```diff
@@ -0,0 +1,284 @@
+        String restoringMessageID;
+
+        DLCBrowsePage dlcBrowsePage = homePage.getDLCBrowsePage();
+        Assert.assertNotNull(dlcBrowsePage.isDLCCreated(),
+                             "DeadLetter Channel not created. " + DLC_TEST_DURABLE_TOPIC);
+        //Testing delete messages
+        DLCContentPage dlcContentPage = dlcBrowsePage.getDLCContent();
+        deletingMessageID = dlcContentPage.deleteFunction();
+
+        Assert.assertTrue(checkMessages(deletingMessageID, DLC_TEST_DURABLE_TOPIC),
+                          "Deleting messages of dead letter channel is unsuccessful.");
+
+        // number of messages received by consumer client before restore function triggered.
+        Long beforeRestoreMessageReceivedCount = consumerClient.getReceivedMessageCount();
+
+        //Testing restore messages
+        restoringMessageID = dlcContentPage.restoreFunction();
+        waitUntilMessagesReceived(9L);
+
+        //Thread sleep until messages sent to DLC after breaching maximum number of retrying
+        AndesClientUtils.sleepForInterval(80000L);
+
+        // number of messages received by consumer client after restore function triggered.
+        Long afterRestoreMessageReceivedCount = consumerClient.getReceivedMessageCount();
+
+        // This assertion will check if consumer client has received messages messages after
+        // restore function triggered from ui. If it receives messages after restore function triggered
+        // this assertion will be success.
+        Assert.assertTrue(beforeRestoreMessageReceivedCount < afterRestoreMessageReceivedCount,
+                          restoringMessageID + " Durable topic message not successfully restored.");
+    }
+
+    /**
+     * This method will wait until given number of messages received by the consumer client.
+     *
+     * @param numberOfMessages this will specify number of messages to wait before proceed
+     *
+     */
+    public void waitUntilMessagesReceived(Long numberOfMessages) {
+
+
+        while (true) {
+            Long count = consumerClient.getReceivedMessageCount();
+            if (count >= numberOfMessages) {
+                break;
+            }
+            AndesClientUtils.sleepForInterval(1000L);
+        }
+    }
+
+
+    /**
+     * Check whether element is present or not
+     *
+     * @param id which element check for its availability
+     * @return availability of the element
+     */
+    public boolean isElementPresent(String id) {
+        return driver.findElements(By.xpath(id)).size() != 0;
+    }
+
+    /**
+     * Search messageID through all messages in the queue
+     *
+     * @param deletingMessageID - Searching messageID
+     * @param queueName             - Searching queue
+     * @return whether messageID available or not
+     */
+    private boolean checkMessages(String deletingMessageID, String queueName) {
+        boolean isSuccessful = true;
+        if (isElementPresent(UIElementMapper.getInstance()
```

Comment:
IMO it's good to keep this method separately. We have to check if element is present before accessing it using xpath. Otherwise it will throw a null point exception. In future if there are new ui tests for dlc we can use this same method to check if is present.